File: clients/health-ios/Sources/HealthGatewayClient/HealthIngestEnvelope.swift
import Foundation

public struct HealthIngestEnvelope: Sendable, Equatable, Codable {
    public let connectorID: String
    public let tenantID: String
    public let nodeID: String
    public let sampleType: String
    public let submittedAt: Date
    public let payloadRevision: Int

    public init(
        connectorID: String,
        tenantID: String,
        nodeID: String,
        sampleType: String,
        submittedAt: Date,
        payloadRevision: Int
    ) {
        self.connectorID = connectorID
        self.tenantID = tenantID
        self.nodeID = nodeID
        self.sampleType = sampleType
        self.submittedAt = submittedAt
        self.payloadRevision = payloadRevision
    }

    // Maps to the ingest envelope JSON contract (backend/api/health.py HealthIngestPayload).
    enum CodingKeys: String, CodingKey {
        case connectorID = "connector_id"
        case tenantID = "tenant_id"
        case nodeID = "node_id"
        case sampleType = "sample_type"
        case submittedAt = "submitted_at"
        case payloadRevision = "payload_revision"
    }
}

File: clients/health-android/src/main/kotlin/io/zen70/healthgateway/HealthIngestEnvelope.kt
package io.zen70.healthgateway

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Ingest envelope submitted to the gateway health ingest endpoint.
 *
 * Field names match the backend ingest contract:
 * backend/api/health.py HealthIngestPayload.
 *
 * [submittedAt] is a Unix epoch second (Long). The backend validates that
 * it falls within ±5 minutes of server time to prevent replay attacks.
 */
@Serializable
data class HealthIngestEnvelope(
    @SerialName("connector_id") val connectorId: String,
    @SerialName("tenant_id") val tenantId: String,
    @SerialName("node_id") val nodeId: String,
    @SerialName("sample_type") val sampleType: String,
    @SerialName("submitted_at") val submittedAt: Long,
    @SerialName("payload_revision") val payloadRevision: Int,
)
